### PR TITLE
[SPARK-28797][DOC] Document DROP FUNCTION statement in SQL Reference.

### PR DESCRIPTION
**Pull request by sandeep-katta**
### What changes were proposed in this pull request?
Add DROP FUNCTION sql description in SQL reference

### Why are the changes needed?
Currently from spark there is no complete sql guide is present, so it is better to document all the sql commands, this jira is sub part of this task.

### Does this PR introduce any user-facing change?
Yes before user cannot find any reference for drop function command in the spark docs.

After Fix:
image

image

image

### How was this patch tested?
tested with jekyll build

### CODE REVIEW

File: docs/sql-ref-syntax-ddl-drop-function.md [hunk 1]
```diff
@@ -19,4 +19,31 @@ license: |
...
+
+### Parameters
+
+### **function_name**
+
+The name of an existing function.
+
+### **TEMPORARY**
+
+If specified, the function created will be local to session. On close of session it will be removed.
```
Comment (dilipbiswal) [inline docs/sql-ref-syntax-ddl-drop-function.md]: This does not sound right. We are talking about create in a drop action ?

File: docs/sql-ref-syntax-ddl-drop-function.md [hunk 2]
```diff
@@ -19,4 +19,31 @@ license: |
...
+
+### **IF EXISTS**
+
+If specified will not throw any exception if no function exists.
```
Comment (dilipbiswal) [inline docs/sql-ref-syntax-ddl-drop-function.md]: How about -
"If specified, no exception is thrown when the function does not exist"

**Comment by dilipbiswal** [comment]
@sandeep-katta Hello, could you please update this by following the convention of existing merged PRS. Thanks a lot.
https://github.com/apache/spark/pull/25529
https://github.com/apache/spark/pull/25525

**Comment by dilipbiswal** [comment]
LGTM
cc @gatorsmile

**Comment by gatorsmile** [comment]
ok to test

File: docs/sql-ref-syntax-ddl-drop-function.md [hunk 3]
```diff
@@ -19,4 +19,64 @@ license: |
...
+-- List user functions
+SHOW user functions;
+  +-------------------+
+  |     function      |
+  +-------------------+
+  | default.test_avg  |
+  +-------------------+
+
+-- Drop Permanent function
+DROP FUNCTION test_avg;
```
Comment (gatorsmile) [inline docs/sql-ref-syntax-ddl-drop-function.md]: Can we document the behavior when there exists a temporary function whose name is also  `test_avg`?

File: docs/sql-ref-syntax-ddl-drop-function.md [hunk 4]
```diff
@@ -19,4 +19,64 @@ license: |
...
+  <dd>If specified, no exception is thrown when the function does not exist.</dd>
+</dl>
+
+### Example
+{% highlight sql %}
+-- Create a permanent function `test_avg`
+CREATE FUNCTION test_avg as 'org.apache.hadoop.hive.ql.udf.generic.GenericUDAFAverage';
+
+-- List user functions
+SHOW user functions;
```
Comment (viirya) [inline docs/sql-ref-syntax-ddl-drop-function.md]: SHOW user functions; -> SHOW USER FUNCTIONS;?

**Comment by gatorsmile** [comment]
We need to show the precedence of temp view and table when they share the same name.

File: docs/sql-ref-syntax-ddl-drop-function.md [hunk 5]
```diff
@@ -59,6 +59,18 @@ SHOW USER FUNCTIONS;
...
+-- Create Temporary function `temp_avg`
+CREATE TEMPORARY FUNCTION temp_avg as 'org.apache.hadoop.hive.ql.udf.generic.GenericUDAFAverage';
+
+-- List user functions
+SHOW USER FUNCTIONS;
+  +-------------------+
+  |     function      |
+  +-------------------+
+  | default.test_avg  |
+  | temp_avg          |
```
Comment (gatorsmile) [inline docs/sql-ref-syntax-ddl-drop-function.md]: Are the functions having the same name?

Comment (sandeep-katta) [reply docs/sql-ref-syntax-ddl-drop-function.md]: yes, one with `permanent` and other with `Temporary`

Comment (dilipbiswal) [reply docs/sql-ref-syntax-ddl-drop-function.md]: @gatorsmile The permanent functions are always shown as database qualified where as the temporary functions are not qualified.

Comment (sandeep-katta) [reply docs/sql-ref-syntax-ddl-drop-function.md]: oops :), sorry I corrected it 

image

**Comment by sandeep-katta** [comment]
@dilipbiswal  @gatorsmile   does any more changes are required for this PR ?

File: docs/sql-ref-syntax-ddl-drop-function.md [hunk 6]
```diff
@@ -19,4 +19,87 @@ license: |
   limitations under the License.
 ---
 
-**This page is under construction**
+### Description
+`DROP FUNCTION` statement drops a temporary or user defined function(UDF). An exception will be thrown if the function does not exist
```
Comment (srowen) [inline docs/sql-ref-syntax-ddl-drop-function.md]: Nit: Add "The" at the start. Also, space before (UDF). What does 'in the system' mean - can we just drop this or make it more specific?

File: docs/sql-ref-syntax-ddl-drop-function.md [hunk 7]
```diff
@@ -19,4 +19,87 @@ license: |
   limitations under the License.
 ---
 
-**This page is under construction**
+### Description
+The `DROP FUNCTION` statement drops a temporary or user defined function (UDF). An exception will
+ be thrown if the function does not exists.
```
Comment (srowen) [inline docs/sql-ref-syntax-ddl-drop-function.md]: exists -> exist

**Review by srowen — APPROVED** [review]

File: docs/sql-ref-syntax-ddl-drop-function.md [hunk 8]
```diff
@@ -19,4 +19,87 @@ license: |
...
+  Error: Error running query:
+  org.apache.spark.sql.catalyst.analysis.NoSuchPermanentFunctionException:
+  Function 'default.test_avg' not found in database 'default'; (state=,code=0)
+
+-- List the functions after dropping, it should list only temporary function
+SHOW USER FUNCTIONS;
+  +-----------+
+  | function  |
+  +-----------+
+  | temp_avg  |
```
Comment (dilipbiswal) [inline docs/sql-ref-syntax-ddl-drop-function.md]: @sandeep-katta Why is it temp_avg here ? Shouldn't it be temporary function `test_avg` ?

File: docs/sql-ref-syntax-ddl-drop-function.md [hunk 9]
```diff
@@ -19,4 +19,87 @@ license: |
...
+  +-----------+
+  
+-- Drop Temporary function
+DROP TEMPORARY FUNCTION IF EXISTS temp_avg;
```
Comment (dilipbiswal) [inline docs/sql-ref-syntax-ddl-drop-function.md]: @sandeep-katta same question :-)

File: docs/sql-ref-syntax-ddl-drop-function.md (hunk 8 above)
Comment (srowen) [reply docs/sql-ref-syntax-ddl-drop-function.md]: Ping @sandeep-katta to fix that part

**Review by srowen — CHANGES_REQUESTED** [review]

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**